wiring/src/spark_wiring_ble.cpp

```diff
@@ -2109,6 +2109,15 @@ int BleLocalDevice::setAdvertisingType(BleAdvertisingEventType type) const {
     return hal_ble_gap_set_advertising_parameters(&advParams, nullptr);
 }
 
+int BleLocalDevice::setAdvertisingPhy(EnumFlags<BlePhy> phy) const {
+    if(phy != BlePhy::CODED && phy != BlePhy::MBPS1) return SYSTEM_ERROR_NOT_SUPPORTED;
+    hal_ble_adv_params_t advParams = {};
+    advParams.size = sizeof(hal_ble_adv_params_t);
```

Comment:
We should also set the version field with `BLE_API_VERSION` just in case for backward compatibility. The BLE HAL may make use of this field to execute code conditionally. Please help also init this field in other wiring APIs those need to set/get HAL struct.

Reply:
The first thing that `BleObject::Broadcaster::setAdvertisingParams()` does is set the version field to `BLE_API_VERSION`, so it seems like if we set it in the Wiring function, it'll get overwritten. 

Are you thinking about changing `setAdvertisingParams()` to only set version if it's not set, that way Wiring APIs can set different values of version? Currently that's not possible.

Reply:
We just set the version for the temporary advertising params in `setAdvertisingParams()`. The internal cached advertising params version can be the latest so that the relevant logic in HAL can always handle the latest params properly, for example, we just introduce the new field `primary_phy` and the BLE HAL should check (although we don't check as of right now) the version and handle this field properly. While the version filed set in an old user application (which means that user built the application against an old Device OS version, but the on-device Device OS has been updated to the latest) may be lower then the version in Device OS, in that case the BLE HAL can check the version field that is passed in by user application and init/tweak some fields' value and then finally update the cached params for later usage. I know that we don't do this for now, but for better practice we should set the version field in wiring just in case we make use of this filed in BLE HAL in the future.

Reply:
There is an example in place: https://github.com/particle-iot/device-os/blob/develop/hal/src/nRF52840/spi_hal.cpp#L479

Reply:
Basically, the version field we set in BLE HAL is to make sure the internal logic comply with the latest struct, while the version field set by user application determines how we exchange the params between BLE HAL and wiring.

Reply:
Ok, I think I have it now.